packages/credentials-interface/src/utils.ts

```diff
@@ -0,0 +1,100 @@
+import { Credential, VerifiableCredential } from './credentials.types';
+import { Presentation } from './presentation.types';
+
+function isObject(obj: unknown): obj is object {
+  return typeof obj === 'object' && obj !== null;
+}
+
+function hasOwnProperty<X extends object, Y extends PropertyKey>(
```

Comment:
Can we use https://lodash.com/docs/4.17.15#has?

Reply:
Probably yes, but why we need external library for such easy function?

Reply:
First it allows to reduce code. We wouldn't need to add `isObject`, `hasOwnedProperty`, `hasRequiredProperties` as they are all implemented in `lodash`. But most important is that it would allow to avoid some tricky errors. For example `isObject` returns true when it is passed function, which is probably not what we want

Reply:
I agree that maybe it will prevent errors, but it won't reduce the code. I've created `hasOwnProperty` and `isObject` mainly for typescript reason (`obj is object`, `obj is X & Record<Y, unknown>`). These functions help me declare types and avoid `@ts-ignore`, bc input object is unknown.

Reply:
Do you mean that tsc doesn't compile? I thought that `@types/lodash` has definitions